[Bug]: PCI device deletion fails

### What happened?

I was trying to remove a PCI device from VirtualMachine object and it failed. 

### What did you expect to happen?

I was expecting this one to be completed without any errors or crash. 

### How can we reproduce it (as minimally and precisely as possible)?

Apply the following example VirtualMachine object:
```
`# This manifest is used to create a Virtual Machine from an existing template.
apiVersion: proxmox.alperen.cloud/v1alpha1
kind: VirtualMachine
metadata:
  name: virtualmachine-sample-clone
spec:
  name: virtualmachine-sample-clone
  # Name of the node where the VM will be created
  nodeName: lowtower
  # Deletion protection is whether to delete VM from Proxmox or not
  deletionProtection: false
  # VM should be started any time found in stopped state
  enableAutoStart: true
  template:
    # Name of the template to be cloned
    name: fedora-template 
    # CPU cores to be allocated to the VM
    cores: 2
    # CPU sockets to be allocated to the VM
    socket: 1
    # Memory to be allocated to the VM
    memory: 4096 # As MB
    # Disk used by the VM
    disk: 
      - storage: local-lvm 
        size: 60 # As GB
        device: scsi0
      - storage: local-lvm 
        size: 20 # As GB
        device: scsi1
    # Network interfaces used by the VM
    network:
      - model: virtio
        bridge: vmbr0
    pciDevices:
      - deviceID: "0000:03:00.2"
        type: "raw"
        primaryGPU: true
        pcie: true
  # Optional field to add additional features to the VM that doesn't included in the spec
  additionalConfig:                                         
    balloon: "0"                                            
    cpu: host                                               
    vga: type=virtio,memory=16   `
```

Then remove the pciDevices and you will face with the error.

### Anything else we need to know?

2025-02-19T17:13:46+01:00       INFO    Reconciling VirtualMachine virtualmachine-sample-clone  {"controller": "virtualmachine", "controllerGroup": "proxmox.alperen.cloud", "controllerKind": "VirtualMachine", "VirtualMachine": {"name":"virtualmachine-sample-clone"}, "namespace": "", "name": "virtualmachine-sample-clone", "reconcileID": "45cdb7bc-7bf8-41e3-a331-9ca2313c3924"}
2025-02-19T17:13:50+01:00       INFO    VirtualMachine virtualmachine-sample-clone has been started     {"controller": "virtualmachine", "controllerGroup": "proxmox.alperen.cloud", "controllerKind": "VirtualMachine", "VirtualMachine": {"name":"virtualmachine-sample-clone"}, "namespace": "", "name": "virtualmachine-sample-clone", "reconcileID": "45cdb7bc-7bf8-41e3-a331-9ca2313c3924"}
2025-02-19T17:13:50+01:00       INFO    Updating network configuration for VirtualMachine virtualmachine-sample-clone
2025-02-19T17:13:50+01:00       INFO    Observed a panic in reconciler: runtime error: index out of range [0] with length 0     {"controller": "virtualmachine", "controllerGroup": "proxmox.alperen.cloud", "controllerKind": "VirtualMachine", "VirtualMachine": {"name":"virtualmachine-sample-clone"}, "namespace": "", "name": "virtualmachine-sample-clone", "reconcileID": "45cdb7bc-7bf8-41e3-a331-9ca2313c3924"}
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 374 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:115 +0x19c
panic({0x101fe13e0?, 0x14000c16018?})
        /opt/homebrew/Cellar/go/1.24.0/libexec/src/runtime/panic.go:787 +0x124
github.com/alperencelik/kubemox/pkg/proxmox.parsePCIConfiguration(0x140002a7f00?)
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/pkg/proxmox/virtualmachine.go:1247 +0x1e4
github.com/alperencelik/kubemox/pkg/proxmox.configureVirtualMachinePCI(0x14000c8a900)
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/pkg/proxmox/virtualmachine.go:1200 +0x9c
github.com/alperencelik/kubemox/pkg/proxmox.ConfigureVirtualMachine(0x14000c8a900)
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/pkg/proxmox/virtualmachine.go:799 +0x38
github.com/alperencelik/kubemox/internal/controller/proxmox.(*VirtualMachineReconciler).UpdateVirtualMachine(0x1400046d500, {0x102116288, 0x14000cccff0}, 0x14000c8a900)
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/internal/controller/proxmox/virtualmachine_controller.go:343 +0x7c
github.com/alperencelik/kubemox/internal/controller/proxmox.(*VirtualMachineReconciler).handleVirtualMachineOperations(0x1400046d500, {0x102116288, 0x14000cccff0}, 0x14000c8a900)
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/internal/controller/proxmox/virtualmachine_controller.go:208 +0xb8
github.com/alperencelik/kubemox/internal/controller/proxmox.(*VirtualMachineReconciler).Reconcile(0x1400046d500, {0x102116288, 0x14000cccff0}, {{{0x0?, 0x140004e5d28?}, {0x140002a65c0?, 0x30?}}})
        /Users/alperencelik/Desktop/Personal/Repos/kubemox/internal/controller/proxmox/virtualmachine_controller.go:138 +0x5d0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x102116288?, {0x102116288?, 0x14000cccff0?}, {{{0x0?, 0x101e0c560?}, {0x140002a65c0?, 0x14000b01608?}}})
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118 +0x80
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x1400028f040, {0x1021162c0, 0x140002eee60}, {0x101f65120, 0x14000c9ccc0})
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314 +0x280
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x1400028f040, {0x1021162c0, 0x140002eee60})
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265 +0x180
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226 +0x70
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 118
        /Users/alperencelik/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:222 +0x3fc

### Kubemox version

<details>

```console
v0.3.0
```

</details>


### Proxmox version

<details>

```console
# On Proxmox target node: 
$ pveversion 
pve-manager/8.2.2/9355359cd7afbae4 (running kernel: 6.8.4-2-pve)
```

</details>

--- a/pkg/proxmox/virtualmachine.go
+++ b/pkg/proxmox/virtualmachine.go
@@ -1390,6 +1390,17 @@ func applyPCIChanges(
 	pcisToAdd, pcisToUpdate, pcisToDelete []proxmoxv1alpha1.PciDevice,
 ) error {
 	getDeviceID := func(pci proxmoxv1alpha1.PciDevice) string {
+		// Use hostpci[N] when deleting; for add/update, use the raw DeviceID
+		for _, del := range pcisToDelete {
+			if del.DeviceID == pci.DeviceID {
+				index, err := getIndexOfPCIConfig(vm.Spec.Name, vm.Spec.NodeName, pci)
+				if err != nil {
+					log.Log.Error(err, "Error getting PCI config index for deletion")
+					return pci.DeviceID
+				}
+				return index
+			}
+		}
 		return pci.DeviceID
 	}
 
@@ -1415,6 +1426,7 @@ func updatePCIConfig(ctx context.Context, vm *proxmoxv1alpha1.VirtualMachine,
 		log.Log.Error(err, "Error getting VM")
 		return err
 	}
+	// Index returns as hostpci[n]
 	index, err := getIndexOfPCIConfig(vmName, nodeName, pci)
 	if err != nil {
 		log.Log.Error(err, "Error getting index of PCI configuration")
@@ -1422,7 +1434,7 @@ func updatePCIConfig(ctx context.Context, vm *proxmoxv1alpha1.VirtualMachine,
 	}
 
 	task, err := VirtualMachine.Config(ctx, proxmox.VirtualMachineOption{
-		Name:  "hostpci" + index,
+		Name:  index,
 		Value: buildPCIOptions(pci),
 	})
 	if err != nil {
@@ -1467,6 +1479,8 @@ func updatePCIConfig(ctx context.Context, vm *proxmoxv1alpha1.VirtualMachine,
 	return nil
 }
 
+// getIndexOfPCIConfig returns the index of the PCI device in the VM configuration
+// Returns as hostpci0, hostpci1, etc.
 func getIndexOfPCIConfig(vmName, nodeName string, pciDevice proxmoxv1alpha1.PciDevice) (string, error) {
 	VirtualMachine, err := getVirtualMachine(vmName, nodeName)
 	if err != nil {

